stdlib/public/core/Codable.swift

```diff
@@ -5510,23 +5510,101 @@ internal struct _DictionaryCodingKey: CodingKey {
   internal let stringValue: String
   internal let intValue: Int?
 
-  internal init?(stringValue: String) {
+  internal init(stringValue: String) {
     self.stringValue = stringValue
     self.intValue = Int(stringValue)
   }
 
-  internal init?(intValue: Int) {
+  internal init(intValue: Int) {
     self.stringValue = "\(intValue)"
     self.intValue = intValue
   }
+
+  fileprivate init(codingKey: CodingKey) {
+    self.stringValue = codingKey.stringValue
+    self.intValue = codingKey.intValue
+  }
+}
+
+/// A type that can be converted to and from a coding key.
+///
+/// With a `CodingKeyRepresentable` type, you can losslessly convert between a
+/// custom type and a `CodingKey` type.
+///
+/// Conforming a type to `CodingKeyRepresentable` lets you opt in to encoding and
+/// decoding `Dictionary` values keyed by the conforming type to and from a keyed
+/// container, rather than encoding and decoding the dictionary as an unkeyed container
+/// of alternating key-value pairs.
```

Comment:
```suggestion
/// Conforming a type to `CodingKeyRepresentable` lets you opt in to encoding
/// and decoding `Dictionary` values keyed by the conforming type to and from
/// a keyed container, rather than encoding and decoding the dictionary as an
/// unkeyed container of alternating key-value pairs.
```